### ISSUE TITLE
Next steps?

### ISSUE THREAD

**Issue by nestanyol** [nestanyol/CoolBeans#28]
@lwjohnst86 
Luke, which steps are we missing for building the R package? :) Any recommendations?

**Comment by lwjohnst86** [comment]
Yes! I plan on looking over what you've done now over the weekend and next week and make some issues, do some reviews and stuff 🤩

**Comment by lwjohnst86** [comment]
- Many of the `test_` files contain functions that are very very similar. Maybe try to consider ways to simplify it so you don't need to repeat the same code throughout them all.
- Many functions output a list of dataframes. It might be a good idea to think how your functions can be created so that only a single dataframe is output. Trying to follow the principle of "Object type as input, same object type as output"  (https://design.tidyverse.org/out-type-stability.html), particularly the "tibble in, tibble out" principle makes things easier to chain together with the pipes. Likewise, if the function takes a list, then the best practice is to return a list.
- Many of the functions need more descriptive Roxygen titles (e.g. `test_knn_regression()` has the title `test_model_knn_regression`, which isn't very descriptive.
- While technically `return()` is used to output the end result of a function, in practice, we don't use it. See more details here: https://style.tidyverse.org/functions.html#return
- I did some PR's with comments, and also accidentally pushed to the main branch, but I have some commits there with some comments as well. 
- Consider naming your functions as verbs (e.g. instead of `preprocess_data()`, call it simply `preprocess()`).
- Some of your arguments' names are shortened `cutoff_subj`, consider keeping the full word instead for clarity of description. 
- It's time to start writing some unit tests! See #37 for some details about that! To run unit testing while in RStudio, use `Ctrl-Shift-T`.
- I've added some infrastructure files to have your code get tested and checked on GitHub, automatically. See #38
- You'll eventually need to create your documentation using `devtools::document()` (Ctrl-Shift-D). To run that, you need to merge #39 first. You'll probably encounter a lot of errors and warnings as you update/fix your documentation within the package.
- Once you've finished with most of these, you can start running local CRAN checks for basic checks of your package. You can do that by running `Ctrl-Shift-E`. But only do that later, after dealing with all of theses :grin:

**Comment by danielaags** [comment]
@lwjohnst86 Nuria told me that I could be a good idea to join the training and testing functions into a single one. Shall we also put that in the to do list?

**Comment by lwjohnst86** [comment]
In general, it's best to keep functions small and very focused/targeted. So keeping the testing and training separate is better than having them together, especially since, looking at their code, they have quite a bit different in them.

**Comment by danielaags** [comment]
I have implemented the suggestions in all the functions. I would like to start working on the unit tests, how shall I move forward? Shall I merge the unit test branch into main and the pull it in my branch (DDLS branch)?
